docs/en_US/Compression/v2_pruning_algo.rst

```diff
@@ -273,6 +274,65 @@ User configuration for ADMM Pruner
 
 .. autoclass:: nni.algorithms.compression.v2.pytorch.pruning.ADMMPruner
 
+Movement Pruner
+---------------
+
+Movement pruner is an implementation of movement pruning.
+This is a pruning by step algorithm, the masks may change during each step.
+Each weight element will be scored by the opposite of the sum of the product of weight and its gradient during each step.
+This means the weight elements moving towards zero will accumulate negative scores, the weight elements moving away from zero will accumulate positive scores.
+The weight elements with low scores will be masked during inference.
+
+The following figure from the paper shows the weight pruning by movement pruning.
+
+.. image:: ../../img/movement_pruning.png
+   :target: ../../img/movement_pruning.png
+   :alt: 
+
+For more details, please refer to `Movement Pruning: Adaptive Sparsity by Fine-Tuning <https://arxiv.org/abs/2005.07683>`__.
+
+Usage
+^^^^^^
+
+.. code-block:: python
+
+   from nni.algorithms.compression.v2.pytorch.pruning import MovementPruner
+   config_list = [{'op_types': ['Linear'], 'op_partial_names': ['bert.encoder'], 'sparsity': 0.9}]
+   pruner = MovementPruner(model, config_list, p_trainer, optimizer, criterion, 10, 3000, 27000)
+   masked_model, masks = pruner.compress()
+
+   # ignore the parameters with `weight_score` in name if you want to finetune with masks
+   optimizer_grouped_parameters = [{
+        "params": [p for n, p in model.named_parameters() if "weight_score" not in n and p.requires_grad]
```

Comment:
What is `weight_score` for? Can we handle this automatically so that user don't have to modify the optimizer manually? Besides, whether `weight_score`  limits our appliable scenario to a specific implement version/repo of transformer?

Reply:
![image](https://user-images.githubusercontent.com/33053116/143365673-3caed073-7c5a-4ee6-83f2-644c3f9cc949.png)

`weight_score` is register in wrapper as a parameter, it is the sum of `- weight * weight_grad`.
It's OK that user directly use `optimizer = Adam(model.named_parameters(), lr=2e-5)`, just some computing resources were wasted. But it's a good idea that we handle this automatically, I will try this.

`weight_score` will not limit our appliable scenario, all module that has `weight` can use this pruner.

Reply:
> Can we handle this automatically so that user don't have to modify the optimizer manually?

fix it